environment-setup/tp1/students/playbook-tp1-users.yml

```diff
@@ -0,0 +1,67 @@
+---
+- hosts: localhost
+  gather_facts: false
+  vars:
+    ansible_ssh_pipelining: true
+    student_groups: "hadoop"
+    hadoop_bin_path: "/home/hadoop/hadoop/bin"
+  vars_files:
+    - students.yml
+  tasks:
+    - name: Ensure hadoop group exists
+      become: true
+      ansible.builtin.group:
+        name: "{{ student_groups }}"
+        state: present
+
+    - name: Add users to group
+      become: true
+      ansible.builtin.user:
+        name: "{{ item.username }}"
+        groups: "{{ student_groups }}"
+        state: present
+        append: true
+        create_home: yes 
+        shell: /bin/bash
+      with_items: "{{ students }}"
+      loop_control:
+        label: "{{ item.username }}"
+    
+    - name: Create .ssh directory for each user
+      become: true
+      file:
+        path: "/home/{{ item.username }}/.ssh"
+        state: directory
+        owner: "{{ item.username }}"
+        group: "{{ student_groups }}"
+        mode: '0700'
+      with_items: "{{ students }}"
+
+    - name: Add public key to authorized_keys
+      become: true
+      copy:
```

Comment:
Please use the [Ansible module](https://docs.ansible.com/ansible/latest/collections/ansible/posix/authorized_key_module.html) for increased safety.